stdlib/importlib/metadata/__init__.pyi

```diff
@@ -61,7 +61,53 @@ class EntryPoint(_EntryPointBase):
         def attr(self) -> str: ...
     if sys.version_info >= (3, 10):
         dist: ClassVar[Distribution | None]
-        def matches(self, **params: Any) -> bool: ...  # undocumented
+        def matches(
+            self,
+            __name: str = ...,
+            __value: str = ...,
+            __group: str = ...,
+            __module: str = ...,
+            __attr: str = ...,
+            __extras: list[str] = ...,
+        ) -> bool: ...  # undocumented
+
+if sys.version_info >= (3, 10):
+    class EntryPoints(list[EntryPoint]):  # use as list is deprecated since 3.10
+        # int argument is deprecated since 3.10
+        def __getitem__(self, item: int | str) -> EntryPoint: ...  # type: ignore[override]
+        def select(
+            self,
+            __name: str = ...,
+            __value: str = ...,
+            __group: str = ...,
+            __module: str = ...,
+            __attr: str = ...,
+            __extras: list[str] = ...,
+        ) -> EntryPoints: ...
+        @property
+        def names(self) -> set[str]: ...
+        @property
+        def groups(self) -> set[str]: ...
+
+    class SelectableGroups(dict[str, EntryPoints]):  # use as dict is deprecated since 3.10
+        @classmethod
+        def load(cls: type[Self], eps: Iterable[EntryPoint]) -> Self: ...
+        @property
+        def groups(self) -> set[str]: ...
+        @property
+        def names(self) -> set[str]: ...
+        @overload
+        def select(self: Self) -> Self: ...  # type: ignore[misc]
+        @overload
+        def select(
+            self,
+            __name: str = ...,
+            __value: str = ...,
+            __group: str = ...,
+            __module: str = ...,
+            __attr: str = ...,
+            __extras: list[str] = ...,
```

Comment:
I don't think these should start with dunder, they are being hidden this way.

Check this example:
```py
from importlib import metadata as importlib_metadata

(mypy,) = importlib_metadata.entry_points().select(name="mypy")
print(mypy)
```

Running `mypy` on it results in:
```
example.py:3: error: No overload variant of "select" of "SelectableGroups" matches argument type "str"
example.py:3: note: Possible overload variants:
example.py:3: note:     def select(self) -> SelectableGroups
example.py:3: note:     def select(self, str = ..., str = ..., str = ..., str = ..., str = ..., List[str] = ...) -> EntryPoints
Found 1 error in 1 file (checked 1 source file)
```

pylance/pyright doesn't suggest arguments either:
![image](https://user-images.githubusercontent.com/12860284/155136918-9d512cfd-d188-45f3-9602-e09a07114e52.png)

But if I remove dunders for all arguments (I think this should apply to all functions/methods in this file) everything seems to work fine:
![image](https://user-images.githubusercontent.com/12860284/155136749-516df3ea-2b1e-479e-8651-90508849e5b5.png)

Reply:
The reason I prefixed these arguments with a dunder is because from what I could recall, according to the conventions, that's how you annotate that an argument is keyword-only. However, I now see that this is actually the convention for positional-only arguments 😅

Reply:
Fixed.